apps/api/src/app/inbox/usecases/notifications-count/notifications-count.usecase.ts

```diff
@@ -43,15 +45,56 @@ export class NotificationsCount {
       throw new BadRequestException('Filtering for unread and archived notifications is not supported.');
     }
 
+    const retentionDate = await this.getRetentionPeriod(command.organizationId);
     const getCountPromises = command.filters.map((filter) =>
-      this.messageRepository.getCount(command.environmentId, subscriber._id, ChannelTypeEnum.IN_APP, filter, {
-        limit: MAX_NOTIFICATIONS_COUNT,
-      })
+      this.messageRepository.getCount(
+        command.environmentId,
+        subscriber._id,
+        ChannelTypeEnum.IN_APP,
+        filter,
+        { limit: MAX_NOTIFICATIONS_COUNT },
+        { $gte: retentionDate }
+      )
     );
 
     const counts = await Promise.all(getCountPromises);
     const result = counts.map((count, index) => ({ count, filter: command.filters[index] }));
 
     return { data: result };
   }
+
+  private async getRetentionPeriod(organizationId: string): Promise<Date> {
```

Comment:
If not mistaken, this code exists elsewhere in the code base, so let's extract it to an API-specific utility.